qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaExternalDBExtension.java

```diff
@@ -0,0 +1,152 @@
+  public static final String DEFAULT_ES_URL = "http://localhost:9200";
+  public static final String DEFAULT_OS_URL = "http://localhost:9200";
+  public static final String DEFAULT_OS_ADMIN_USER = "admin";
+  public static final String DEFAULT_OS_ADMIN_PW = "yourStrongPassword123!";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaExternalDBExtension.class);
+  private final DatabaseType databaseType;
+  private final List<AutoCloseable> closeables = new ArrayList<>();
+  private final NewCamundaTestApplication testApplication;
+  private String testPrefix;
+
+  public CamundaExternalDBExtension() {
+    this(new NewCamundaTestApplication());
+
+    closeables.add(testApplication);
+    testApplication
+        .withBrokerConfig(cfg -> cfg.getGateway().setEnable(true))
+        .withRecordingExporter(true);
+  }
+
+  public CamundaExternalDBExtension(final NewCamundaTestApplication testApplication) {
+    this.testApplication = testApplication;
+    // resolve active database and exporter type
+    final String property = System.getProperty(PROP_CAMUNDA_IT_DATABASE_TYPE);
+    databaseType =
+        property == null ? DatabaseType.ES : DatabaseType.valueOf(property.toUpperCase());
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    LOGGER.info("Starting up Camunda instance, with {}", databaseType);
+    final Class<?> testClass = context.getRequiredTestClass();
+    testPrefix = testClass.getSimpleName().toLowerCase();
+
+    switch (databaseType) {
+      case ES -> testApplication.withElasticsearchSupport(DEFAULT_ES_URL, testPrefix);
+      case OS ->
+          testApplication.withOpensearchSupport(
+              DEFAULT_OS_URL, testPrefix, DEFAULT_OS_ADMIN_USER, DEFAULT_OS_ADMIN_PW);
+      case RDBMS -> testApplication.withRdbmsExporter();
+      default -> throw new RuntimeException("Unknown exporter type");
+    }
+    testApplication.start();
+    testApplication.awaitCompleteTopology();
+
+    injectFields(testClass, null, ModifierSupport::isStatic);
+  }
+
+  private void injectFields(
+      final Class<?> testClass, final Object testInstance, Predicate<Field> predicate) {
+    predicate = predicate.and(field -> field.getType() == CamundaClient.class);
+    for (final Field field : testClass.getDeclaredFields()) {
+      try {
+        if (predicate.test(field)) {
+          field.setAccessible(true);
+          field.set(testInstance, createCamundaClient());
+        }
+      } catch (final Exception ex) {
+        throw new RuntimeException(ex);
+      }
+    }
+  }
+
+  @Override
+  public void afterAll(final ExtensionContext context) {
+    if (databaseType == DatabaseType.ES || databaseType == DatabaseType.OS) {
+      final URI deleteEndpoint = URI.create(String.format("%s/%s*", DEFAULT_ES_URL, testPrefix));
+      final HttpRequest httpRequest = HttpRequest.newBuilder().DELETE().uri(deleteEndpoint).build();
```

Comment:
:question: Is this _really_ deleting everything? Indices, templates, component templates, aliases, etc.? All things listed here? https://github.com/camunda/camunda/issues/26043

Not a blocker though - in fact, why not make use of the purge feature once it's done? Right now it's only implemented for RDBMS, but we will have it for all DBs soon, and it will purge your C8 runtime (so log stream, snapshots, etc.) and the history storage (e.g. ES, RDBMS). We can make it as a TODO.

Reply:
See: https://github.com/camunda/camunda/issues/25861

Reply:
> Not a blocker though - in fact, why not make use of the purge feature once it's done?

Because its not done

Reply:
It was a rhetorical question :smile: I hope I would know if it was done or not :smile:

Reply:
Yeah i guess i would create an issue for it. I feel todos are more an antipattern and not really useful.

But tbh not sure who would take care of it.

Reply:
That sounds so evil 💀

Reply:
![dark side](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExZnR4dnM2NWU4dmFiYTN5enU1N253ZGZzaTZjbGwwM2dzNTA1dnBqbCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/GIIC4jmmUlXZS/giphy.gif)

Reply:
Thanks. I created a follow up to use this feature in integration tests https://github.com/camunda/camunda/issues/27264